src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -256,28 +313,37 @@ internal class ModelCheckingStrategy(
                 switchableThreads(iThread).random(interleavingFinishingRandom)
             }
 
-        fun isSwitchPosition() = executionPosition in switchPositions
+        fun isSwitchPosition() = currentSwitchCrashChoice == 0 && executionPosition in switchPositions
+        fun isCrashPosition() = currentSwitchCrashChoice == 1 && executionPosition in crashPositions
 
         /**
-         * Creates a new execution position that corresponds to the current switch point.
+         * Creates a new execution position that corresponds to the current switch/crash point.
          * Unlike switch points, the execution position is just a gradually increasing counter
          * which helps to distinguish different switch points.
          */
-        fun newExecutionPosition(iThread: Int) {
+        private fun newExecutionPosition(iThread: Int, isSwitch: Boolean) {
+            if (exploreCrashNode == isSwitch) return
```

Comment:
This seems like a bug. Whether or not nodes in interleaving tree are initialized should not change execution position numbering, otherwise wrong positions will be saved in a node initialization

Reply:
I've rearranged the lines, so increment is always executed now.